### PR TITLE
Fix for missing index vars in `for-in`, add tests

### PR DESCRIPTION
**Pull request by wcjohnson**
Fixes https://github.com/lightscript/lightscript/issues/30

### CODE REVIEW

File: babylon-to-espree/toAST.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 var source;
+var t = require("babel-types");
```
Comment (rattrayalex) [inline babylon-to-espree/toAST.js:2]: unfortunately, creating raw nodes this way removes `loc` information, which ESLint generally needs... hence the super-kludgy cloning that's been going on so far.

**Review by rattrayalex — COMMENTED** [review]
I don't think we can use babel-types here b/c we need `loc` info 😕 and if this runs after the "babel-to-estree" conversion, then it'll break on that too (can't immediately remember if it does or not)

**Comment by wcjohnson** [comment]
Ah, right, good point. Will also add a failing test case to catch that.

**Comment by rattrayalex** [comment]
awesome, thanks.

it might make sense to add a helper function like you made for source locs
in the compiler, if you wanna do that you're welcome to

**Comment by wcjohnson** [comment]
@rattrayalex 

I fixed this to generate source location information. Because they will be used here and in the compiler, as well as in several other projects I'm working on, I factored the ast node location tools to an npm module: https://www.npmjs.com/package/ast-loc-utils

I can inline them here if you don't want the extra dependency but ultimately I think factoring that code out for reuse is correct.

**Review by rattrayalex — COMMENTED** [review]
Cool – will these be converted to ESTree nodes from Babel nodes?

**Comment by wcjohnson** [comment]
This is all happening in `lscNodesToBabelNodes`:  https://github.com/wcjohnson/lightscript-eslint/blob/8baae58ad76f17cba3145bf419b61ea793bdf750/babylon-to-espree/toAST.js#L53

So they will be further transformed to ESTree nodes afterward.

**Comment by rattrayalex** [comment]
Awesome, thanks! Sorry I wasn't able to look into that myself, have been super busy lately. 

If you feel like refactoring other parts of the file to use the same system, you're more than welcome to, but no obligation. 

Nice work as always!

**Pull request merged**